keyboards/tau4/keymaps/default/keymap.c

```diff
@@ -0,0 +1,135 @@
+        KC_BSPC, KC_LCTL, KC_LGUI, KC_LALT, MO(_LOWER), KC_SPC, KC_SPC, MO(_RAISE), KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT),
+
+    [_NUMPAD] = LAYOUT(
+        KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_P7, KC_P8,   KC_P9,   KC_PSLS,
+        KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_P4, KC_P5,   KC_P6,   KC_PAST,
+        KC_TRNS, TG(_NUMPAD), KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_P1, KC_P2,   KC_P3,   KC_PENT,
+        KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_P0, KC_PDOT, KC_PMNS, KC_PPLS),
+
+    [_LOWER] = LAYOUT(
+        KC_VOLD, KC_EXLM,       KC_AT,    KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN,      KC_RPRN,       KC_INS,
+        KC_NLCK, KC_F7,         KC_F8,    KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_LCBR, KC_RCBR, KC_UNDS,      KC_PLUS,       LSFT(KC_NUHS),
+        KC_TRNS, LSFT(KC_NUBS), KC_GRAVE, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, LSFT(KC_GRV), LSFT(KC_NUBS), KC_TRNS,
+        KC_TRNS, KC_TRNS,       KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_PGDN,      KC_PGUP,       KC_END),
+
+    [_RAISE] = LAYOUT(
+        KC_VOLU, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
+        KC_CAPS, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_LBRC, KC_RBRC, KC_MINS, KC_EQL,  KC_NUHS,
+        KC_TRNS, KC_PSCR, KC_SLCK, KC_PAUS, KC_APP,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_GRV,  KC_NUBS, KC_TRNS,
+        KC_TAB,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MSTP, KC_MPRV, KC_MPLY, KC_MNXT),
+
+    [_ADJUST] = LAYOUT(
+        KC_TRNS, RGB_TOG,     RGB_MOD, RGB_HUD, RGB_HUI,  RGB_SAD,  RGB_SAI, RGB_VAD, RGB_VAI, KC_TRNS, KC_TRNS, RESET,
+        KC_TRNS, RGB_M_P,     RGB_M_B, RGB_M_R, RGB_M_SW, RGB_M_SN, RGB_M_K, RGB_M_X, RGB_M_G, RGB_M_T, KC_TRNS, KC_TRNS,
+        KC_TRNS, TG(_NUMPAD), KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS,     KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (get_highest_layer(update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST))) {
+    case _RAISE:
+        setrgb (0x6E,  0x19, 0x17, (LED_TYPE *)&led[7]);
+        break;
+    case _LOWER:
+        setrgb (0x15,  0x50, 0x4C, (LED_TYPE *)&led[7]);
+        break;
+    case _ADJUST:
+        setrgb (0x7E,  0x7B, 0x71, (LED_TYPE *)&led[7]);
+        break;
+    case _NUMPAD:
+        setrgb (0x00,  0x00, 0xFF, (LED_TYPE *)&led[7]);
+        break;
+    default: //  for any other layers, or the default layer
+        setrgb (0x00,  0x00, 0x00, (LED_TYPE *)&led[7]);
+        break;
+    }
+
+    rgblight_set();
+
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
```

Comment:
The rgb light layers feature does the same thing, but "better". 
https://docs.qmk.fm/#/feature_rgblight?id=lighting-layers

The advantage to this is that it's set in the core code. Specifically, after animations are sent, but before the LED array is sent to the strip.  This prevents flickering issues, and is more performant, in theory. 

```suggestion
const rgblight_segment_t PROGMEM my_raise_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, 1, 202, 110}
);
const rgblight_segment_t PROGMEM my_lower_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, 125, 188, 80}
); 
const rgblight_segment_t PROGMEM my_adjust_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, 32, 25, 127}
);
const rgblight_segment_t PROGMEM my_numpad_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, HSV_BLUE}
);
const rgblight_segment_t PROGMEM my_default_layer[] = RGBLIGHT_LAYER_SEGMENTS(
    {7, 1, HSV_OFF}
);

const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
    my_default_layer,
    my_raise_layer,
    my_lower_layer,
    my_adjust_layer,
    my_numpad_layer
);

void keyboard_post_init_user(void) {
    rgblight_layers = my_rgb_layers;
}

layer_state_t layer_state_set_user(layer_state_t state) {
    state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);

    rgblight_set_layer_state(_QWERTY, layer_state_cmp(state, _QWERTY));
    rgblight_set_layer_state(_RAISE, layer_state_cmp(state, _RAISE));
    rgblight_set_layer_state(_LOWER, layer_state_cmp(state, _LOWER));
    rgblight_set_layer_state(_ADJUST, layer_state_cmp(state, _ADJUST));
    rgblight_set_layer_state(_NUMPAD, layer_state_cmp(state, _NUMPAD));

    return state;
}
```

Reply:
I've made commits for all these changes, and it seems to build and work fine on my keyboard. 

The lighting layers is a much better solution, it's now working with animations so thanks for that :+1: .

Reply:
I'm glad to hear it, and you are very welcome!